pkg/alertmanager/amcfg.go

```diff
@@ -506,7 +506,7 @@ func (cb *configBuilder) convertRoute(in *monitoringv1alpha1.Route, crKey types.
 }
 
 // convertReceiver converts a monitoringv1alpha1.Receiver to an alertmanager.receiver
-func (cb *configBuilder) convertReceiver(ctx context.Context, in *monitoringv1alpha1.Receiver, crKey types.NamespacedName) (*receiver, error) {
+func (cb *configBuilder) convertReceiver(ctx context.Context, in *monitoringv1alpha1.Receiver, globalCfg *globalConfig, crKey types.NamespacedName) (*receiver, error) {
```

Comment:
since `configBuilder` already stores the config, I'd avoid passing another argument. The only (?) precaution needed is in `initializeFromAlertmanagerConfig()` where it needs to initialize the `cfg.Global`  field before converting the other fields.